File: java/OneLoop.java
import java.util.*;

//Write a function that takes in a list of integers as an argument and returns true if it contains two integers that are the same, false otherwise.
public class OneLoop {

    public static boolean twoIntegers(List<Integer> list) {
        for (int initial = 0; initial < list.size(); initial++) { // For every value in the list...
            int numberOfEncounters = 0; // reset counter
            for (int other = 0; other < list.size(); other++) { // ..check every value in the list
                if (list.get(initial) == list.get(other)) { // check against it, to see if it is the same
                    numberOfEncounters++;
                } // nOE should increase to 1 for all values, but if it appears twice, there is a
                  // double of an occurance
                if (numberOfEncounters == 2) {
                    return true;
                }
            }
        }
        return false;

    }

    public static void main(String[] args) {
        List<Integer> list1 = new ArrayList<Integer>(); // declare a list
        list1.add(12);
        list1.add(15);
        list1.add(32);
        list1.add(89);
        list1.add(33);

        List<Integer> list2 = new ArrayList<Integer>(); // declare a list
        list2.add(21);
        list2.add(15);
        list2.add(32);
        list2.add(15);
        list2.add(30);

        boolean List1 = twoIntegers(list1);
        boolean List2 = twoIntegers(list2);

        System.out.println("Double in List 1?: " + List1);
        System.out.println("Double in List 2?: " + List2);

    }

}

File: python/one_loop.py
def one_loop(l: list) -> bool:
    for i in range(len(l)):
        nOE = 0
        for j in range(len(l)):
            if l[i] == l[j]:
                nOE=nOE + 1
        if nOE == 2:
            return True
    return False    

if __name__ == "__main__":
    List1 = one_loop([1,2,3,4,5,6])
    List2 = one_loop([23,91,54,29,23,30])

    print(List1)
    print(List2)